Merge `lyftga` branch into `master`

mistercrunch opened the pull request:
Hey I wanted to start the conversation around this and assess how doable this is..

### CATEGORY

Choose one

- [x] crazy large branch merge
- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
People at Lyft have been working on stabilizing a branch using an alternative workflow / process. Many bug fixes needed by the community exist here only.

### TEST PLAN
This branch has been deployed in production at Lyft

betodealmeida committed:
Show tooltip with time frame (#6979)

betodealmeida committed:
Fix time filter control (#6978)

khtruong committed:
feat: add ability to change font size in big number (#7003)

* Add ability to change font sizes in Big Number

* rename big number to header

* Add comment to clarify font size values

betodealmeida committed:
Allow LIMIT to be specified in parameters (#7052)

betodealmeida committed:
Revert PR #6933 (#7162)

khtruong committed:
[Lyft-GA] Enable color consistency in a dashboard (#7135)

* Enable color consistency in a dashboard

Moved actions, minor UI, allowed dashboard copy

Fix linting errors

Undo unintentional change

Updated and added unit tests

Fail quietly if package has not been updated

Fail quietly on dashboard copy if package is old

* Update packages

* Remove unnecessary code

* Addressed Grace's comments

* Small fix for item key

* Reset chart's color during exploration

* Do not reset chart form data when exploring chart

betodealmeida committed:
Fix race condition when fetching results in SQL Lab (#7198)

* Fix race condition when fetching results in SQL Lab

* Fix lint

khtruong committed:
Change number format default (#7137)

* Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

* Fix the white background shown in SQL editor on drag (#7021)

This PR sets the background-color css property on `.ace_scroller` instead of `.ace_content` to prevent the white background shown during resizing of the SQL editor before drag ends.

* Show tooltip with time frame (#6979)

* Fix time filter control (#6978)

* Enhancement of query context and object. (#6962)

* added more functionalities for query context and object.

* fixed cache logic

* added default value for groupby

* updated comments and removed print

(cherry picked from commit d5b9795f87f79fa2c41e144ffc00fd9586be7657)

* [fix] /superset/slice/id url is too long (#6989)


(cherry picked from commit 6a4d507ab607b01ed324cb3341b71c6fb2cb5c97)

* [WIP] fix user specified JSON metadata not updating dashboard on refresh (#7027)


(cherry picked from commit cc58f0e661044e95c7c86d0da8d77a0a6640efe7)

* feat: add ability to change font size in big number (#7003)

* Add ability to change font sizes in Big Number

* rename big number to header

* Add comment to clarify font size values

* Allow LIMIT to be specified in parameters (#7052)

* [fix] Cursor jumping when editing chart and dashboard titles (#7038)


(cherry picked from commit fc1770f7b79a4d8815b646b46390fabf190c3815)

* Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

* [db-engine-spec] Aligning Hive/Presto partition logic (#7007)


(cherry picked from commit 05be86611785fef2904992e4e7d31dce23f1c51b)

* [fix] explore chart from dashboard missed slice title (#7046)


(cherry picked from commit a6d48d4052839286aec725d51303b3b2bf6e8dd4)

* fix inaccurate data calculation with adata rolling and contribution (#7035)


(cherry picked from commit 0782e831cd37f665a2838119d87c433269f1b36b)

* Adding warning message for sqllab save query (#7028)


(cherry picked from commit ead3d48133e7e1ab8b91d51e561544a544b4eaad)

* [datasource] Ensuring consistent behavior of datasource editing/saving. (#7037)

* Update datasource.py

* Update datasource.py

(cherry picked from commit c771625f1068d3a7f41e6bced14b0cbdbf9962cc)

* [csv-upload] Fixing message encoding (#6971)


(cherry picked from commit 48431ab5b9375a94c5262a0336d9c69e5f01a3ac)

* [sql-parse] Fixing LIMIT exceptions (#6963)


(cherry picked from commit 3e076cb60b385e675ed1c9a8053493375e43370b)

* Adding custom control overrides (#6956)

* Adding extraOverrides to line chart

* Updating extraOverrides to fit with more cases

* Moving extraOverrides to index.js

* Removing webpack-merge in package.json

* Fixing metrics control clearing metric

(cherry picked from commit e6194051f486e42922dc4e34a861f4490c1062fc)

* [sqlparse] Fixing table name extraction for ill-defined query (#7029)


(cherry picked from commit 07c340cf8203f13222f16efad1e55e202deb1865)

* [missing values] Removing replacing missing values (#4905)


(cherry picked from commit 61add606ca16a6ba981ccde864b121f5464b697a)

* [SQL Lab] Improved query and results tabs rendering reliability (#7082)

closes #7080

(cherry picked from commit 9b58e9f4920ef424e5b545dcbb4726e22bed5982)

* Fix filter_box migration PR #6523 (#7066)

* Fix filter_box migration PR #6523

* Fix druid-related bug

(cherry picked from commit b210742ad24d01ca05bc58ca3342c90e301fe073)

* SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

* [forms] Fix handling of NULLs

(cherry picked from commit e83a07d3dfda350cc44041cb6cbaec4510887902)

* handle null column_name in sqla and druid models

(cherry picked from commit 2ff721ae072b8d69c5cabddc3e1a388a596b1b6f)

* Use metric name instead of metric in filter box (#7106)


(cherry picked from commit 003364e74ea70cad1a4a6e784933fe8bef4c78ec)

* Bump python lib croniter to an existing version (#7132)

Package maintainers should really never delete packages, but it appears
this happened with croniter and resulted in breaking our builds.

This PR bumps to a more recent existing version of the library

(cherry picked from commit 215ed392a11598eac228f57341dbfd232cf770e3)

* Revert PR #6933 (#7162)

* [bugfix] SQL Lab 'Filter Results' doesn't stick (#7104)

When using a "Search Results" criteria, the subset of rows that match
the criteria get displayed. While this the filter is applied, if another
query is run, the filter is still active, but not displayed in the input
text box. After this change, the state of the input box sticks after
subsequent queries.

(cherry picked from commit d5e8d663dc63ee9b99b3e7bb4e91512fa32ed1e0)

* Injectable statsd client (#7138)

* Add ability to inject statsd client; some py test/reqs updates

- Updated the metrics logger to allow construction with an existing
statsd client, so that it can be configured by external systems or libs.
- added requirements to requirements-dev.txt which are needed to run
  tests-eg coverage, nose
- removed dependency on mock lib, it is in python stdlib now
- updated tox.ini to remove the now-superfluous deps

* add license to test file, and remove blank line at EOF

(cherry picked from commit ba19a62412aba27d9c38ec7013f13a976cb6c054)

* [Lyft-GA] Enable color consistency in a dashboard (#7135)

* Enable color consistency in a dashboard

Moved actions, minor UI, allowed dashboard copy

Fix linting errors

Undo unintentional change

Updated and added unit tests

Fail quietly if package has not been updated

Fail quietly on dashboard copy if package is old

* Update packages

* Remove unnecessary code

* Addressed Grace's comments

* Small fix for item key

* Reset chart's color during exploration

* Do not reset chart form data when exploring chart

* Fix double scroll bars when content of sql result table overflows horizontally (#7168)

The PR substracts the scrollbar height from the height of the container of the react virtualized table so we don't see double scrollbars.

(cherry picked from commit 7ffcabd5c08bbe6aa022e66590c0ce805ae13f08)

* Change number format default

* Use smart formatter instead

* fix merge issues

* Use SMART_NUMBER

khtruong committed:
Update necessary packages (#7205)

khtruong committed:
fix: [lyftga] pass the latest deckgl props (#7208)

xtinec committed:
chore: revert bignumber.js patch for charts and reapply the original bignumber.js change to SQL editor in an opt-in fashion (#7210)

* revert: bignumber patch to prevent rendering regression in charts

* Revert "Fix rendering regression from the introduction of bignumber (#6937)"

* fix: consume the bignumber.js reversion -- apply the bignumber conversion in `actions/sqlLab.js` where it is needed (aka opt into bignumber.js).

* Revert "Fix deck.gl form data (#6953)" b/c formData now returns snake_cased properties for deck vizzes.

betodealmeida committed:
Fetch charts with GET to benefit from browser cache and conditional requests (#7032)

* Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

* Fix the white background shown in SQL editor on drag (#7021)

This PR sets the background-color css property on `.ace_scroller` instead of `.ace_content` to prevent the white background shown during resizing of the SQL editor before drag ends.

* Show tooltip with time frame (#6979)

* Fix time filter control (#6978)

* Enhancement of query context and object. (#6962)

* added more functionalities for query context and object.

* fixed cache logic

* added default value for groupby

* updated comments and removed print

(cherry picked from commit d5b9795f87f79fa2c41e144ffc00fd9586be7657)

* [fix] /superset/slice/id url is too long (#6989)


(cherry picked from commit 6a4d507ab607b01ed324cb3341b71c6fb2cb5c97)

* [WIP] fix user specified JSON metadata not updating dashboard on refresh (#7027)


(cherry picked from commit cc58f0e661044e95c7c86d0da8d77a0a6640efe7)

* feat: add ability to change font size in big number (#7003)

* Add ability to change font sizes in Big Number

* rename big number to header

* Add comment to clarify font size values

* Allow LIMIT to be specified in parameters (#7052)

* [fix] Cursor jumping when editing chart and dashboard titles (#7038)


(cherry picked from commit fc1770f7b79a4d8815b646b46390fabf190c3815)

* Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

* [db-engine-spec] Aligning Hive/Presto partition logic (#7007)


(cherry picked from commit 05be86611785fef2904992e4e7d31dce23f1c51b)

* [fix] explore chart from dashboard missed slice title (#7046)


(cherry picked from commit a6d48d4052839286aec725d51303b3b2bf6e8dd4)

* fix inaccurate data calculation with adata rolling and contribution (#7035)


(cherry picked from commit 0782e831cd37f665a2838119d87c433269f1b36b)

* Adding warning message for sqllab save query (#7028)


(cherry picked from commit ead3d48133e7e1ab8b91d51e561544a544b4eaad)

* [datasource] Ensuring consistent behavior of datasource editing/saving. (#7037)

* Update datasource.py

* Update datasource.py

(cherry picked from commit c771625f1068d3a7f41e6bced14b0cbdbf9962cc)

* [csv-upload] Fixing message encoding (#6971)


(cherry picked from commit 48431ab5b9375a94c5262a0336d9c69e5f01a3ac)

* [sql-parse] Fixing LIMIT exceptions (#6963)


(cherry picked from commit 3e076cb60b385e675ed1c9a8053493375e43370b)

* Adding custom control overrides (#6956)

* Adding extraOverrides to line chart

* Updating extraOverrides to fit with more cases

* Moving extraOverrides to index.js

* Removing webpack-merge in package.json

* Fixing metrics control clearing metric

(cherry picked from commit e6194051f486e42922dc4e34a861f4490c1062fc)

* [sqlparse] Fixing table name extraction for ill-defined query (#7029)


(cherry picked from commit 07c340cf8203f13222f16efad1e55e202deb1865)

* [missing values] Removing replacing missing values (#4905)


(cherry picked from commit 61add606ca16a6ba981ccde864b121f5464b697a)

* [SQL Lab] Improved query and results tabs rendering reliability (#7082)

closes #7080

(cherry picked from commit 9b58e9f4920ef424e5b545dcbb4726e22bed5982)

* Fix filter_box migration PR #6523 (#7066)

* Fix filter_box migration PR #6523

* Fix druid-related bug

(cherry picked from commit b210742ad24d01ca05bc58ca3342c90e301fe073)

* SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

* [forms] Fix handling of NULLs

(cherry picked from commit e83a07d3dfda350cc44041cb6cbaec4510887902)

* handle null column_name in sqla and druid models

(cherry picked from commit 2ff721ae072b8d69c5cabddc3e1a388a596b1b6f)

* Use metric name instead of metric in filter box (#7106)


(cherry picked from commit 003364e74ea70cad1a4a6e784933fe8bef4c78ec)

* Bump python lib croniter to an existing version (#7132)

Package maintainers should really never delete packages, but it appears
this happened with croniter and resulted in breaking our builds.

This PR bumps to a more recent existing version of the library

(cherry picked from commit 215ed392a11598eac228f57341dbfd232cf770e3)

* Revert PR #6933 (#7162)

* Add decorator for etag cache

* Fetch charts with GET

* Small fixes

* Fix typo

* Compute correct cache key; fix logging

* Check perms on cached response

* Revert change

* If perms fail, return naked response

* Fix lint

* Compute cache key from all form data

* Pass extra_filters in GET request

* Fix pylint

* Fix flake8

* Use ETags even if no cache is set

* Handle adhoc filters

* Raise in debug mode

* Rename actions

* Fix integration tests

* Do POST request on new charts

* Set extra/adhoc filters only in GET requests

* Raise if check_perms fails

* Refactor auth

* Fix flake8

* Fix js unit tests

* Fix js unit tests that fail in lyftga

* Fix js

* Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

* Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

* SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

* Add decorator for etag cache

* Fetch charts with GET

* Small fixes

* Fix typo

* Compute correct cache key; fix logging

* Check perms on cached response

* Revert change

* If perms fail, return naked response

* Fix lint

* Compute cache key from all form data

* Pass extra_filters in GET request

* Fix pylint

* Fix flake8

* Use ETags even if no cache is set

* Handle adhoc filters

* Raise in debug mode

* Rename actions

* Fix integration tests

* Do POST request on new charts

* Set extra/adhoc filters only in GET requests

* Raise if check_perms fails

* Refactor auth

* Fix flake8

* Fix js unit tests

* Fix js unit tests that fail in lyftga

* Fix js

* Fix bad merge

* Use far future when max_age=0

betodealmeida committed:
Allow bigint to be used as time column in Presto (#7218)

mistercrunch committed:
Merge branch 'master' into lyftga

khtruong committed:
feat: [lyftga] add tooltips to deck.gl viz types (#7206)

* Add tooltips to deck.gl viz types

* Handle text overflow

* Use .css file instead of .less

* fix: add tooltip component

* Add tooltips to deck.gl viz types

* Handle text overflow

* Use .css file instead of .less

* fix: add tooltip component

* fix: casing, types, and keys

betodealmeida committed:
Celery task for warming up cache (#7148)

* Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

* Fix the white background shown in SQL editor on drag (#7021)

This PR sets the background-color css property on `.ace_scroller` instead of `.ace_content` to prevent the white background shown during resizing of the SQL editor before drag ends.

* Show tooltip with time frame (#6979)

* Fix time filter control (#6978)

* Enhancement of query context and object. (#6962)

* added more functionalities for query context and object.

* fixed cache logic

* added default value for groupby

* updated comments and removed print

(cherry picked from commit d5b9795f87f79fa2c41e144ffc00fd9586be7657)

* [fix] /superset/slice/id url is too long (#6989)


(cherry picked from commit 6a4d507ab607b01ed324cb3341b71c6fb2cb5c97)

* [WIP] fix user specified JSON metadata not updating dashboard on refresh (#7027)


(cherry picked from commit cc58f0e661044e95c7c86d0da8d77a0a6640efe7)

* feat: add ability to change font size in big number (#7003)

* Add ability to change font sizes in Big Number

* rename big number to header

* Add comment to clarify font size values

* Allow LIMIT to be specified in parameters (#7052)

* [fix] Cursor jumping when editing chart and dashboard titles (#7038)


(cherry picked from commit fc1770f7b79a4d8815b646b46390fabf190c3815)

* Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

* [db-engine-spec] Aligning Hive/Presto partition logic (#7007)


(cherry picked from commit 05be86611785fef2904992e4e7d31dce23f1c51b)

* [fix] explore chart from dashboard missed slice title (#7046)


(cherry picked from commit a6d48d4052839286aec725d51303b3b2bf6e8dd4)

* fix inaccurate data calculation with adata rolling and contribution (#7035)


(cherry picked from commit 0782e831cd37f665a2838119d87c433269f1b36b)

* Adding warning message for sqllab save query (#7028)


(cherry picked from commit ead3d48133e7e1ab8b91d51e561544a544b4eaad)

* [datasource] Ensuring consistent behavior of datasource editing/saving. (#7037)

* Update datasource.py

* Update datasource.py

(cherry picked from commit c771625f1068d3a7f41e6bced14b0cbdbf9962cc)

* [csv-upload] Fixing message encoding (#6971)


(cherry picked from commit 48431ab5b9375a94c5262a0336d9c69e5f01a3ac)

* [sql-parse] Fixing LIMIT exceptions (#6963)


(cherry picked from commit 3e076cb60b385e675ed1c9a8053493375e43370b)

* Adding custom control overrides (#6956)

* Adding extraOverrides to line chart

* Updating extraOverrides to fit with more cases

* Moving extraOverrides to index.js

* Removing webpack-merge in package.json

* Fixing metrics control clearing metric

(cherry picked from commit e6194051f486e42922dc4e34a861f4490c1062fc)

* [sqlparse] Fixing table name extraction for ill-defined query (#7029)


(cherry picked from commit 07c340cf8203f13222f16efad1e55e202deb1865)

* [missing values] Removing replacing missing values (#4905)


(cherry picked from commit 61add606ca16a6ba981ccde864b121f5464b697a)

* [SQL Lab] Improved query and results tabs rendering reliability (#7082)

closes #7080

(cherry picked from commit 9b58e9f4920ef424e5b545dcbb4726e22bed5982)

* Fix filter_box migration PR #6523 (#7066)

* Fix filter_box migration PR #6523

* Fix druid-related bug

(cherry picked from commit b210742ad24d01ca05bc58ca3342c90e301fe073)

* SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

* [forms] Fix handling of NULLs

(cherry picked from commit e83a07d3dfda350cc44041cb6cbaec4510887902)

* handle null column_name in sqla and druid models

(cherry picked from commit 2ff721ae072b8d69c5cabddc3e1a388a596b1b6f)

* Use metric name instead of metric in filter box (#7106)


(cherry picked from commit 003364e74ea70cad1a4a6e784933fe8bef4c78ec)

* Bump python lib croniter to an existing version (#7132)

Package maintainers should really never delete packages, but it appears
this happened with croniter and resulted in breaking our builds.

This PR bumps to a more recent existing version of the library

(cherry picked from commit 215ed392a11598eac228f57341dbfd232cf770e3)

* Revert PR #6933 (#7162)

* Celery worker for warming up cache

* Remove testing changes

* Add documentation

* Fix lint

* WIP dashboard filters

* Use new cache so it works with dashboards

* Add more unit tests, fix old ones

* Fix flake8 and docs

* Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

* Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

* SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

* Celery worker for warming up cache

* Remove testing changes

* Add documentation

* Fix lint

* WIP dashboard filters

* Use new cache so it works with dashboards

* Add more unit tests, fix old ones

* Fix flake8 and docs

* Fix bad merge and pylint

khtruong committed:
fix: add check for polygon tooltips (#7226)

betodealmeida committed:
Fix time grain after revert (#7228)

khtruong committed:
fix: give a more accurate time (#7221)

mistercrunch committed:
[filter_box] allow empty filters list (#7220)

in some cases, people want a time filter only on filter box, without
specifying dimensions (filters), this allows that

thomaswang-shift committed:
increment version number for nvd3 (#7232)

* increment version number for nvd3

* fix: update package-lock.json

* fix: update package-lock.json

* remove package-lock.json created by accident

betodealmeida committed:
Fix for tagging backend (#7240)

thomaswang-shift committed:
increment version numbers (#7245)

* increment version numbers

* increment version numbers

* increment nvd3

betodealmeida committed:
Fix spinning in Welcome page (#7247)

betodealmeida committed:
Fix partition query in Presto (#7250)

betodealmeida committed:
Add link to documentation (#7251)

betodealmeida committed:
Improve cache (#7227)

* Improve cache

* Improve code, add docs

* Simplify flow

* Update docstring

betodealmeida committed:
Fix refresh frequency (#7248)

* Fix refresh frequency

* Fix unit tests

khtruong committed:
fix: Handle rendering a single point (#7256)

* fix: Handle rendering a single point

* fix: typo

betodealmeida committed:
Pass viz type to GET requests (#7255)

* Pass viz type to GET requests

* Fix integration tests

betodealmeida committed:
Bump pydruid to 0.5.2 (#7260)

betodealmeida committed:
Add extra bucket when max outside range (#7264)

betodealmeida committed:
Revert "Fix issues around Database permissions (#7009)" (#7267)

This reverts commit f5274a9c7f437dcf9359f8dc5fdd056c23bf03b8.

betodealmeida committed:
Do not add tag to non-saved objects (#7266)

khtruong committed:
chore: update big number package (#7268)

john-bodley commented:
In terms of "doable" shouldn't we strive to get this branch merged into `master` seeing as i) it contains a number of bug fixes, and ii) there's merit in having various institutions deploying official releases as opposed to forking. 

It may be worthwhile discussing whether we should cut 0.32 before or after merging this branch.

xtinec commented:
@john-bodley @mistercrunch  the `lyftga` branch was a pragmatic choice for us to reduce surface area of code changes in order to commit to an internal GA date with confidence. This branch is supposed to be short-lived (like a feature branch). It is not meant to be a long-lived fork.

We did a soft GA release on 4/9 and found a few issues (mostly in the new features we added) we still need to address prior to our hard release date of 4/23. We plan to continue to check fixes to lyftga till EOD 4/23.

The plan is to take the payload to master after 4/23 and delete the lyftga branch. Does this sound reasonable? If so, we can discuss logistics for who/how/when to bring the payload to master?

khtruong committed:
chore: update legacy packages (#7293)

mistercrunch commented:
Personally I think this is pretty disruptive. Master should be trunk. Now we have a laundry list of things that should have been in `master` that are only in `lyftga`, for an extended & growing period of time. 

Are there things in `lyftga` we do not want into `master`? If not, can we just merge `lyftga` now and periodically until its end of life?

khtruong committed:
fix: time series table (#7302)

* fix: time series table

* fix: add default value for label

* fix: use prop values if defined

* fix: revert CollectionControl changes

khtruong committed:
fix: propagate color mapping from dashboard to charts (#7289)

* fix: propagate color map from dashboard to chart

* fix: handle cache scenario

* fix: give default values

mistercrunch committed:
Fix control validation handling (#7231)

Fixes a series of unexpected things around control validation.

* when a chart opens in a state where a control is invalid, it still
  runs the query, and sometimes gets stuck in what appears to be a 'running'
  state. After this change, no query is run, and a warning is displayed
  in the chart panel body, just like any other error would
* validation used to be done in the <Control> component and alter the
  redux store as it went. Clearly this is not the right approach, now
  validation occurs on loading the initial redux state, as well as in
  the reducer when controls are changed
* currently, when going from a invalid control state to a valid one
  (user addresses what is needed), it auto-triggers a query which can be
  unexpected. After this change, the error message disappears, and the
  "Run Query" overlay gets displayed
* when changing viz type, it's common to get new validation
  errors, and currently when that occurs it will still go ahead and run
  a query with invalid inputs, which often results in errors
  that are not well handled, since much of the logic
  assumes control-validated input.
* prettier control validation messages

(cherry picked from commit a3212eba5df95bca834d8d6d98c11d522d9172f3)
